libirods_rule_engine_plugin-elasticsearch.cpp

```diff
@@ -183,6 +98,92 @@ namespace
 	std::string metadata_index_policy;
 	std::string metadata_purge_policy;
 
+	auto send_http_request(http::verb _verb,
+	                       const std::string_view _target,
+	                       const std::string_view _body = "") -> http_response
+	{
+		// TODO These host/port checks should happen at plugin startup.
+		// There's no point paying for these on every http request.
+
+		for (auto&& host : config->hosts) {
+			if (host.empty()) {
+				rodsLog(LOG_ERROR, "%s: empty service URL.", __func__);
+				continue;
+			}
+
+			namespace urls = boost::urls;
+
+			urls::result<urls::url_view> result = urls::parse_uri(host);
+			if (!result) {
+				rodsLog(LOG_ERROR, fmt::format("{}: could not parse service URL [{}].", __func__, host).c_str());
+				continue;
+			}
+
+			namespace net = boost::asio;
+			using tcp = net::ip::tcp;
+
+			try {
+				net::io_context ioc;
+
+				tcp::resolver resolver{ioc};
+				beast::tcp_stream stream{ioc};
+
+				const auto results = resolver.resolve(result->host(), result->port());
+				stream.connect(results);
+
+				http::request<http::string_body> req{_verb, _target, 11};
+				req.set(http::field::host, boost::asio::ip::host_name());
+				req.set(http::field::user_agent, "iRODS Indexing Plugin/4.3.1");
```

Comment:
Should this be a generated string? Maybe use CMake definitions?

Reply:
Absolutely.

Forgot to capture that. Thanks.

I may tweak this depending on what the common format is for the user agent string.

Reply:
Done.